lib/ecto/changeset.ex

```diff
@@ -1419,6 +1422,7 @@ defmodule Ecto.Changeset do
       |> unique_constraint(:email)
 
   """
+  @spec unique_constraint(t, atom, options) :: t
```

Comment:
Please use `Keyword.t` throughout. Using access is an implementation detail mostly, does not imply we want to support all access based types. ;)